test/controllers/books_controller_test.rb

```diff
@@ -0,0 +1,62 @@
+require "minitest_helper"
+
+class BooksControllerTest < ActionController::TestCase
+  def setup
+    @book = Book.create!(:title => "SICP", :author => "Harold Abelson, Gerald Jay Sussman, Julie Sussman",
+                         :name => "Jazz", :email => "Jazzy@gmail.com", :comment => "" )
+  end
+
+  def test_index
+    get :index
+    assert_response :success
+    assert_equal("Lending Library",    assigns(:title))
+    assert_match @book.title,          @response.body
+  end
+
+  def test_new
+    get :new
+    assert_response :success
+
+    assert_select "input#book_title"
+    assert_select "input#book_author"
+    assert_select "input#book_name"
+    assert_select "input#book_email"
+  end
+
+  def test_create
+    @testbook = Book.create!(:title => "The Pragmatic Programmer", :author => "Andrew Hunt, David Thomas",
+                             :name => "Morgan", :email => "morgan@gmail.com", :comment => "Good book" )
+
+    assert_difference('Book.count', 1) do
+      post :create, book: {
+                             :title => "Winnie The Pooh",
+                             :author => "A. A. Milne",
+                             :name => "Tigger",
+                             :email => "tigger@gmail.com",
+                             :comment => "Favorite book"
+                            }
+    end
+
+    assert_redirected_to books_path
+    get :index
```

Comment:
I think there is a `follow_redirect!` method or something that will let you get around manually going to the index.